spec/controllers/projects/projects_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+
+RSpec.describe Projects::ProjectsController, type: :controller, feature: :projects do
+  before do
+    allow(controller).to receive(:site_name).and_return('SITE')
+  end
+
+  describe 'GET show' do
+    let(:project) { FactoryBot.create(:project) }
+    let(:ability) { Object.new.extend(CanCan::Ability) }
+
+    before do
+      allow(controller).to receive(:current_ability).and_return(ability)
+    end
+
+    context 'with a logged in user who can read the project' do
+      let(:user) { FactoryBot.create(:user) }
+
+      before do
+        ability.can :read, project
+        get :show, params: { id: project.id }, session: { user_id: user.id }
```

Comment:
We normally in other specs we add `before { session[:user_id] = user.id }` instead of setting the session when calling the request. This can be changed when we add more specs to this context.